website/content/docs/integrations/consul-integration.mdx

```diff
@@ -38,58 +37,143 @@ service discovery to connect tasks to other services deployed across your
 cluster. Nomad integrates with Consul to provide service discovery and
 monitoring.
 
-To configure a job to register with service discovery, please see the
+To configure a job to register with service discovery, please refer to the
 [`service` job specification documentation][service].
 
 ## Dynamic Configuration
 
-Nomad's job specification includes a [`template` block](/nomad/docs/job-specification/template)
-that utilizes a Consul ecosystem tool called [Consul Template](https://github.com/hashicorp/consul-template). This mechanism creates a convenient way to ship configuration files
-that are populated from environment variables, Consul data, Vault secrets, or just
-general configurations within a Nomad task.
+Nomad's job specification includes a [`template` block][] that uses a Consul
+ecosystem tool called [Consul Template][]. This mechanism creates a convenient
+way to ship configuration files that are populated from environment variables,
+Consul data, Vault secrets, or just general configurations within a Nomad task.
+
+For more information on Nomad's template block and how it leverages Consul
+Template, please see the [`template` job specification documentation][].
+
+## Using Nomad Workload Identity with Consul
+
+Starting in Nomad 1.7, Nomad clients will use a task or service's [Workload
+Identity][] to authenticate to Consul and obtain a Consul token specific to the
+service or task.
```

Comment:
I only wanted to keep the version number here because we've got specific text talking about both the current and legacy workflows. Once we remove the legacy workflow in 1.9, I'm :+1: for removing it here.